packages/react-scheduler/src/ReactScheduler.js

```diff
@@ -64,12 +52,24 @@ const localDate = Date;
 // This initialization code may run even on server environments
 // if a component just imports ReactDOM (e.g. for findDOMNode).
 // Some environments might not have setTimeout or clearTimeout.
+// However, we always expect them to be defined on the client.
 // https://github.com/facebook/react/pull/13088
 const localSetTimeout =
   typeof setTimeout === 'function' ? setTimeout : (undefined: any);
 const localClearTimeout =
   typeof clearTimeout === 'function' ? clearTimeout : (undefined: any);
 
+// We don't expect either of these to necessarily be defined,
+// but we will error later if they are missing on the client.
+const localRequestAnimationFrame =
+  typeof requestAnimationFrame === 'function'
+    ? requestAnimationFrame
+    : (undefined: any);
+const localCancelAnimationFrame =
+  typeof cancelAnimationFrame === 'function'
+    ? cancelAnimationFrame
+    : (undefined: any);
```

Comment:
This module throws if this is undefined, right? Maybe we should use an invariant?

Reply:
Didn't want to spend the bytes.

I think it actually used an invariant in the past, and Sebastian wanted to relax that to a warning.

Reply:
I guess in sync mode this module is never used so an invariant would be harsh